### DISCUSSION TITLE
High CPU usage i3bar

### DISCUSSION THREAD

**Discussion by flexd** [regolith-linux/regolith-desktop#683] [Q&A] [answered]
Hi there!

I'm having a weird issue where i3bar is using about 50% cpu all the time, possibly after running a while so it may be some sort of leak. The uptime on this machine is currently 26 days, but it has been idle the past 3 weeks since I have been on vacation.
I'm on Ubuntu 20.04.4 LTS.
Hoping you guys have some suggestions here, this is getting a bit annoying. I can temporarily fix the problem by rebooting, but then eventually I end up in the same situation.

```
poll([{fd=5, events=POLLIN|POLLOUT}], 1, -1) = 1 ([{fd=5, revents=POLLOUT}])
writev(5, [{iov_base="\213\32\7\0\1\0\0\0\5\0@\3\n\n\16\16\24\24\377\377\0\0\0\0\0\17\30\0\213\32\7\0"..., iov_len=2300}], 1) = 2300
recvmsg(5, {msg_namelen=0}, 0)          = -1 EAGAIN (Resource temporarily unavailable)
epoll_wait(3, [{EPOLLIN, {u32=7, u64=4294967303}}], 64, 59743) = 1
read(7, ",[{\"separator_block_width\":35,\"m"..., 1024) = 219
read(7, 0x555d5ed18d2b, 805)            = -1 EAGAIN (Resource temporarily unavailable)
^Cstrace: Process 1944299 detached
```
This is what it's continuously spamming when looking at the process in strace, It looks like it's trying to poll something that is no longer here or unavailable somehow?

``` 
% sudo strace -c -p 1944299                                                                                                                                                                  ~
strace: Process 1944299 attached
^Cstrace: Process 1944299 detached
% time     seconds  usecs/call     calls    errors syscall
------ ----------- ----------- --------- --------- ----------------
 24.85    0.004795           1      4627                epoll_wait
 24.67    0.004761           0      9240      4620  read
 24.50    0.004728           1      4620                writev
 13.31    0.002569           0      4620                poll
 12.66    0.002443           0      4628      4621  recvmsg
------ ----------- ----------- --------- --------- ----------------
100.00    0.019296                 27735      9241 total

% i3bar --version                                                                                                                                                                            ~
i3bar version 4.20-non-git © 2010 Axel Wagner and contributors

% sudo dpkg -l | grep regolith-desktop                                                                                                                                                       ~
ii  regolith-desktop-standard                  2.123-1focal                                                              amd64        metapackage for standard Regolith desktop environment

```

Edit: I actually rebooted the machine now, and i3bar is immediately back doing the same thing.

One thing to mention maybe, Xorg is using 17% cpu and i3xrocks is using 15%

Edit2: err, this may have something to do with this

```
Aug 01 11:26:45 oslh-00198-l regolith.desktop[2461392]: /bin/sh: 1: /usr/share/i3xrocks/net-traffic: not found
Aug 01 11:26:45 oslh-00198-l regolith.desktop[3529]: <3>[net-traffic] Command '/usr/share/i3xrocks/$BLOCK_NAME' not found or missing dependency

```

Looks like the correct path is /usr/share/i3xrocks/scripts/net-traffic, I have i3xrocks-net-traffic installed via apt

**Comment by flexd** [comment, marked as answer]
I just updated /etc/regolith/i3xrocks/conf.d/01_setup to point to the correct path and all is now well. Not sure how I ended up with this problem. I did originally have Ubuntu 18.04 on this machine, and probably also in the process upgraded from Regolith 1.x to 2.x, so maybe a config file was not updated in that upgrade?

**Comment by kgilmer** [comment]
Having your bar consume 50% of your CPU certainly would be annoying!  I'm glad you got it figured out.  It's possible that you got bitten by a bug we resolved in 2.0: (the following is debian/apt policy I believe) when a package installs a file into the `/etc` tree, the files are not removed *unless* you `purge` the package from your system.  The intention is to prevent losing any custom configuration files you may have there, but it wrecks a lot of Regolith configuration if in `/etc` because we use `conf.d` style directories.  So, from your error trace above it appears that the net-traffic script was removed (it's in `/usr`) but the i3xrocks config file that references it was not cleaned up upon upgrade.  The fix added in 2.0 was to move all Regolith configuration files to the `/usr` tree.

You may want to poke around your `/etc/regolith` directory if you see other weirdness.  Here's what I have (Regolith 2.1 on Debian Bookworm):

```
$ tree /etc/regolith/
/etc/regolith/
├── i3
│   └── config
├── picom
│   └── config
└── version
```
